src/vs/workbench/electron-browser/main.contribution.ts

```diff
@@ -230,7 +230,12 @@ let workbenchProperties: { [path: string]: IJSONSchema; } = {
 		'type': 'boolean',
 		'description': nls.localize('closeOnFileDelete', "Controls if editors showing a file should close automatically when the file is deleted or renamed by some other process. Disabling this will keep the editor open as dirty on such an event. Note that deleting from within the application will always close the editor and that dirty files will never close to preserve your data."),
 		'default': true
-	}
+	},
+	'workbench.touchBar.enabled': {
```

Comment:
@viztor the setting should only be enabled on macOS, not all platforms. I also suggest to move it into the existing keyboard category:

![image](https://user-images.githubusercontent.com/900690/34099721-967535ca-e3e0-11e7-8c04-e44c63ffa516.png)

E.g. call it `keyboard.touchBar.enabled`